python/lsst/pipe/base/graph.py

```diff
@@ -75,15 +114,15 @@ def __init__(self, taskDef, quanta):
 
 
 class QuantumGraph(list):
-    """QuantumGraph is a sequence of QuantumGraphNodes objects.
+    """QuantumGraph is a sequence of `QuantumGraphTaskNodes` objects.
```

Comment:
I've commented much more extensively above on other possibilities for how to implement `QuantumGraph`, but regardless of what we decide there, I'd prefer to have it not inherit from `list`; inheritance brings a lot of operations that don't obviously make sense for this class, including mutators that make it very easy to violate any invariants the class has (I don't know if we treat the ordering of tasks as an invariant right now, but I think we should at least leave the option of doing that in the future).

Reply:
Original implementation indeed relied on ordering of this list and it made sense with the old CmdLineFwk. With the addition of the new methods it makes less sense now. 
One more point for having per-task list of Quanta is that it allows doing topological sort at task level rather that at individual Quantum level.

Reply:
If we treat topological ordering as a class invariant established (or checked) at construction, we could keep that advantage for the single-preflight-invocation construction pattern.